src/thunderbots/software/util/parameter/generate_parameters.py

```diff
@@ -0,0 +1,291 @@
+    :param group_name: The name of the current group to add to
+    :type param_info: dict
+    :type file_pointer: file
+    :type group_name: str
+
+    """
+    # iterate through keys
+    for key in param_info.keys():
+
+        # if type key is found, create a parameter
+        if "type" in param_info[key]:
+            try:
+                parameter = param_info[key]
+
+                # write parameter to file
+                file_pointer.write(
+                    constants.CFG_PARAMETER.format(
+                        namespace=group_name if group_name is not None else "gen",
+                        name=key,
+                        type=constants.CFG_TYPE_MAP[parameter["type"]],
+                        description=parameter["description"],
+                        default=parameter["default"],
+                        quote="\"" if parameter["type"] in constants.QUOTE_TYPES else "",
+                        min_val=parameter["min"] if parameter["type"] in constants.RANGE_TYPES else None,
+                        max_val=parameter["max"] if parameter["type"] in constants.RANGE_TYPES else None
+                    ))
+
+            except KeyError as kerr:
+                print constants.AUTOGEN_FAILURE_MSG
+                sys.exit('Required key missing from config: {}, please check parameter {}'.format(
+                    kerr, key))
+
+        # if there is no current group, a new namespace must be created
+        elif group_name is None:
+            file_pointer.write(constants.CFG_NEW_NAMESPACE.format(
+                namespace=key
+            ))
+            __cfg_gen(param_info[key], file_pointer, group_name=key)
+
+        # if there already is a group, add to that namespace
+        else:
+            file_pointer.write(constants.CFG_SUB_NAMESPACE.format(
+                namespace=group_name,
+                sub_namespace=key
+            ))
+            __cfg_gen(param_info[key], file_pointer, group_name=key)
+
+#######################################################################
+#                            CPP Generator                            #
+#######################################################################
+
+
+def generate_header_cpp(param_info, output_path):
+    """Takes the input dictionary from the parsed yaml
+    and generates the respective cfg files needed
+
+    :param param_info: Return value of load_configuration, dict containing params
+    :param output_path: Where the cfgs should be saved to
+    :type param_info: dict
+    :type output_path: str
+
+    """
+    # open files
+    dynamic_parameters_h = open(output_path+constants.DYNAMIC_PARMETERS_HEADER, 'w')
+    dynamic_parameters_cpp = open(output_path+constants.DYNAMIC_PARMETERS_CPP, 'w')
+
+    # write the header to .h .cpp file
+    dynamic_parameters_h.write(constants.H_HEADER)
+    dynamic_parameters_cpp.write(constants.CPP_HEADER)
+
+    # generate cfg str used for topic subsription
```

Comment:
``` suggestion
   # generate cfg str used for topic subscribtion
```

Reply:
Oh man, really sorry for all these spelling mistakes, need to setup spell check on vim I completely forgot to do that

Reply:
Ahahah, all good, easy enough for me to make suggestions